### PR TITLE
Fixed issue 10305 Embedded Jetty server fails to start when requests path contains not existed directory

### LINKED ISSUE #10305: Embedded Jetty server fails to start when requests log path in RequestLogWriter contains not existed directory
Embedded Jetty server fails to start with IOException and confusing error message when requests log path settings in RequestLogWriter contains not existed directory: java.io.IOException: Cannot write log directory ...

The IOException with message is thrown from the class RolloverFileOutputStream.

path: root/jetty-util/src/main/java/org/eclipse/jetty/util/RolloverFileOutputStream.java
package org.eclipse.jetty.util;
...
public class RolloverFileOutputStream extends FilterOutputStream
...
    /* ------------------------------------------------------------ */
    private synchronized void setFile()
        throws IOException
    {
        // Check directory
        File file = new File(_filename);
        _filename=file.getCanonicalPath();
        file=new File(_filename);
        File dir= new File(file.getParent());
        if (!dir.isDirectory() || !dir.canWrite())
            throw new IOException("Cannot write log directory "+dir);
...

Also the class RolloverFileOutputStream does not try to create directory for requests log file.
I encountered the bug while testing my example application with Jetty 11.0.15.
I looked at source code of Jetty 12.0.0 and could see that the bug exists there also.

### PR DESCRIPTION
**Pull request by OlexYarm**
Changed error message when requests log path in RequestLogWriter contains not existed directory.

### CODE REVIEW

**Comment by joakime** [comment]
@OlexYarm make sure you have a signed ECA, otherwise this PR cannot be merged (it is strictly forbidden by Eclipse Legal to merge anything written by someone without an ECA on file)

See https://github.com/eclipse/jetty.project/blob/jetty-10.0.x/CONTRIBUTING.md#eclipse-contributor-agreement

File: jetty-util/src/test/java/org/eclipse/jetty/util/RolloverFileOutputStreamTest.java [hunk 1]
```diff
@@ -181,6 +181,47 @@ public void testMidnightRolloverCalcSydneyDSTEnd()
         assertSequence(midnight, expected);
     }
 
+    @Test
```
Comment (lachlan-roberts) [inline jetty-util/src/test/java/org/eclipse/jetty/util/RolloverFileOutputStreamTest.java:187]: If the directory does exist and no exception is thrown we will never enter the `catch` block and the test will pass.

so maybe something like this:
```java
@Test
public void testMissingDirectory()
{
    String templateString = "missingDir/test-rofos-yyyy_mm_dd.log";
    Throwable error;
    try (RolloverFileOutputStream ignored = new RolloverFileOutputStream(templateString))
    {
        throw new IllegalStateException();
    }
    catch (Throwable t)
    {
        error = t;
    }

    assertNotNull(error);
    assertThat(error, instanceOf(IOException.class));
    assertThat(error.getMessage(), containsString("Log directory does not exist."));
}
```

**Review by lachlan-roberts — CHANGES_REQUESTED** [review]
The change to `RolloverFileOutputStream` looks good, but I think this test can be simplified.

File: jetty-util/src/test/java/org/eclipse/jetty/util/RolloverFileOutputStreamTest.java (hunk 1 above)
Comment (OlexYarm) [reply jetty-util/src/test/java/org/eclipse/jetty/util/RolloverFileOutputStreamTest.java:187]: I simplified test method as suggested.

**Review by lachlan-roberts — APPROVED** [review]
LGTM, thanks for the PR

**Comment by OlexYarm** [comment]
Thanks for the suggestions and help, Lachlan.

Alex Y

**Pull request merged**